### PR TITLE
Build the cdac as part of clr.tools instead of as a prereq to the runtime native build

### LINKED ISSUE #104158: [cdacreader] Simplify build order and dependencies
The original `src/native/managed` infrastructure was developed with the idea that we need to build `libcdacreader.so`/`cdacreader.dll` (or `libcdacreader.a`/`cdacreader.a` before static library support was abandoned) ahead of the main CoreCLR build so that we can link to the shared library during the CoreCLR build (specifically during the DAC build).

But we don't do that now.  Instead the DAC shared library dynamically loads the cDAC using `dlopen`/`LoadLibrary`

That means we don't have a true build-time dependency between the cDAC and the rest of the CoreCLR build.  So we don't need to hook it in via `runtime-prereqs.proj`.  We can add it as a separate subset or as a `ProjectReference` some place else.  The only requirements are:
1. Must be available before we package up the runtime
2. Must be available before we create Core_Root for runtime tests
3. Must be available before we run library tests

---

This is also a first step for any plan to publish the cdacreader separately from the runtime (which is a long-term goal of the cdac effort: it should be possible to distribute and use cdacreader out of band)

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes https://github.com/dotnet/runtime/issues/104158

This will also allow us to take a dependency on native components from the cdacreader project when necessary without too much difficulty (ie. native unwinder)

### CODE REVIEW

**Review by AaronRobinsonMSFT — COMMENTED** [review]
I like the idea. I defer to @elinor-fung for confirmation this is the right place.

**Comment by MichalStrehovsky** [comment]
Did you consider making it it's own subset or folding into a new unified subset with existing linuxdac subset? Adding new ILC-compiled things noticeably slows down the build. (I already have cdac commented out locally to speed up my builds and wanted to do some PR about it at some point; this PR looks like a good general direction.)

**Comment by am11** [comment]
We should be able to further simplify the duplicate props https://github.com/am11/runtime/commit/7bba405edeb745ea120809f4b6d6e0059c8f5121.

**Comment by jkoritzinsky** [comment]
> Did you consider making it it's own subset or folding into a new unified subset with existing linuxdac subset? Adding new ILC-compiled things noticeably slows down the build. (I already have cdac commented out locally to speed up my builds and wanted to do some PR about it at some point; this PR looks like a good general direction.)

I considered putting it into its own subset (as tools.cdacreader or something like that), but I didn't want to break the dev workflow of the people who work on it regularly. I don't like putting it into `linuxdac` as the people working on cDAC (and also really most everyone) doesn't need to build the cross-OS dacs for their regular workflows.

If the people who work on cDAC regularly (all tagged for reviewing this PR) are okay with it being in a `tools` subset, I'll happily move it there.

File: eng/Subsets.props [hunk 1]
```diff
@@ -380,6 +381,10 @@
       Test="true" Category="clr" Condition="'$(DotNetBuildSourceOnly)' != 'true' and '$(NativeAotSupported)' == 'true'"/>
   </ItemGroup>
 
+  <ItemGroup Condition="$(_subset.Contains('+tools.cdacreader+'))">
+    <ProjectToBuild Include="$(RepoRoot)\src\native\managed\compile-native.proj" Category="tools" />
```
Comment (am11) [inline eng/Subsets.props:385]: nit: match line 389's style
```suggestion
    <ProjectToBuild Include="$(SharedNativeRoot)managed\compile-native.proj" Category="tools" />
```

**Review by elinor-fung — APPROVED** [review]

**Pull request merged**